lochness/mindlamp/__init__.py

```diff
@@ -38,46 +92,95 @@ def sync(Lochness: 'lochness.config',
-            crc_dst = lochness.crc32file(dst)
-            if crc_dst != crc_src:
-                logger.warn(f'file has changed {dst}')
-                lochness.backup(dst)
-                logger.debug(f'saving {dst}')
+    # how many days of data from current time, default past 10 days
+    days_to_check = get_days_to_pull(Lochness)
+
+    # current time (ct) in UTC
+    ct_utc = datetime.now(pytz.timezone('UTC'))
+
+    # set the cut off point as UTC 00:00:00.00
+    ct_utc_00 = ct_utc.replace(hour=0, minute=0, second=0, microsecond=0)
+
+    for days_from_ct in range(days_to_check):
+        # n day before current time
+        time_utc_00 = ct_utc_00 - timedelta(days=days_from_ct)
+        time_utc_00_ts = time.mktime(time_utc_00.timetuple()) * 1000
+        time_utc_24 = time_utc_00 + timedelta(hours=24)
+        time_utc_24_ts = time.mktime(time_utc_24.timetuple()) * 1000
+
+        # date string to be used in the file name
+        date_str = time_utc_00.strftime("%Y_%m_%d")
+
+        # Extra information for future version
+        # study_id, study_name = get_study_lamp(LAMP)
+        # subject_ids = get_participants_lamp(LAMP, study_id)
+        subject_id = subject.mindlamp[f'mindlamp.{subject.study}'][0]
+        logger.debug(f'Mindlamp {subject_id} {date_str} data pull - start')
+
+        # pull data from mindlamp
+        begin = time.time()
+        activity_dicts = get_activity_events_lamp(
+                LAMP, subject_id,
+                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
+
+        sensor_dicts = get_sensor_events_lamp(
+                LAMP, subject_id,
+                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
+        end = time.time()
+        logger.debug(f'Mindlamp {subject_id} {date_str} data pull - complete')
+
+        # set destination folder
+        # dst_folder = tree.get('mindlamp', subject.general_folder)
+        dst_folder = tree.get('mindlamp',
+                              subject.protected_folder,
+                              processed=False,
+                              BIDS=Lochness['BIDS'])
+
+        # store both data types
+        for data_name, data_dict in zip(['activity', 'sensor'],
+                                        [activity_dicts, sensor_dicts]):
+
+            dst = os.path.join(
+                    dst_folder,
+                    f'{subject_id}_{subject.study}_{data_name}_'
+                    f'{date_str}.json')
+
+            # separate out audio data from the activity dictionary
+            if data_name == 'activity' and data_dict != []:
+                sound_dst = os.path.join(
+                        dst_folder,
+                        f'{subject_id}_{subject.study}_{data_name}_'
+                        f'{date_str}_sound.mp3')
+                data_dict = get_audio_out_from_content(data_dict, sound_dst)
+
+            jsonData = json.dumps(
+                data_dict,
+                sort_keys=True, indent='\t', separators=(',', ': '))
```

Comment:
Shouldn't indent be an integer?

```python
dump(obj, fp, *, skipkeys=False, ensure_ascii=True, check_circular=True, allow_nan=True, cls=None, indent=None, separators=None, default=None, sort_keys=False, **kw)

    If ``indent`` is a non-negative integer, then JSON array elements and
    object members will be pretty-printed with that indent level. An indent
    level of 0 will only insert newlines. ``None`` is the most compact
    representation.
```

I usually use [indent level 3](https://gist.github.com/tashrifbillah/fd9381e1434f4c99b66d48ba0688bca6#file-editjson-py-L27) for pretty printing.

Reply:
Updated to a `int` 3.

Reply:
26aa7f1